lib/sycamore/sycamore/docset.py

```diff
@@ -726,6 +726,21 @@ def custom_filter(doc: Document) -> bool:
         filtered = Filter(self.plan, f=f, **resource_args)
         return DocSet(self.context, filtered)
 
+    def filter_elements(self, f: Callable[[Element], bool], **resource_args) -> "DocSet":
+        """
+        Applies the given filter function to each element in each Document in this DocsSet.
+
+        Args:
+            f: A Callable that takes an Element and returns True if the element should be retained.
+        """
+
+        def process_doc(doc: Document) -> Document:
+            new_elements = filter(f, doc.elements)
+            doc.elements = new_elements
```

Comment:
I've had some issues using `filter` with it giving me a 'filter' object which is an iterable but not a list (lazily evaluated)... mostly typing and serializability things. Might be bc I was trying to chain a bunch of them together, idk. Anyway, would suggest wrapping in `list( . )` or just writing out the comprehension

Reply:
It's interesting that mypy didn't catch this, but I'll switch to a comprehension.